### PR TITLE
Review

### PR DESCRIPTION
**Pull request by jszwedko**
Doing this just for the sake of an easier code review of initial code, this should be closed later.

### CODE REVIEW

File: css/waiting.css [hunk 1]
```diff
@@ -0,0 +1,105 @@
+@import url('https://fonts.googleapis.com/css?family=Open+Sans|Roboto|Droid+Serif');
```
Comment (jszwedko) [inline css/waiting.css:1]: Nitpicks: whitespace and formatting in this file is a little inconsistent. Maybe run it through a CSS formatter?

File: index.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<?php
+header("Cache-Control: no-store, no-cache, must-revalidate");
+header("Cache-Control: post-check=0, pre-check=0", false);
+header("Pragma: no-cache");
+header("Last-Modified: " . gmdate("D, d M Y H:i:s") . " GMT");
+header("Expires: Sat, 26 Jul 1997 05:00:00 GMT");
+
+$config = include('config.php');
+date_default_timezone_set("US/Pacific");
```
Comment (jszwedko) [inline index.php:9]: I assume this matches the timezone set on the `digitalservices.sfgov.org` server?

File: index.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+$theDate = date("F j, Y H:i:s", mktime($serverHour, $serverMinute, $serverSecond, $serverMonth, $serverDay, $serverYear));
+$theTime = strtotime($theDate);
+
+if($theTime > $launchTime) {
+  header('Location:'.$config['redirect']);
+} else {
+  include('wait.php');
+}
+
+?>
```
Comment (jszwedko) [inline index.php:31]: I think we should drop `?>`s in solely PHP files as they tend to cause issues with whitespace (e.g. a newline) finding its way after the closing tag.

File: js/waiting.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+(function() {
+  var s = (new Date()).getTime();
+  var tmp = new moment.duration(moment(new Date(launchDate).getTime())-s);
```
Comment (jszwedko) [inline js/waiting.js:3]: Not crazy about this variable name 😄

File: js/waiting.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    url: 'time.php',
+    data: {
+      t: (new Date()).getTime()
+    },
+    success: function(resp) {
+      var e = (new Date()).getTime();
+      var h = JSON.parse(resp);
+      var u = (new Date(h.month + ' ' + h.mday + ', ' + h.year + ' ' + h.hours + ':' + h.minutes + ':' + h.seconds)).getTime() - (e - s);
+      var i = (new Date(launchDate).getTime());
+      if(u > i) {
```
Comment (jszwedko) [inline js/waiting.js:18]: Same here. I'm all for short variable names when the context makes it clear what they are, but I find this a little hard to follow.

File: js/waiting.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  var countdownStr = function(tms) {
+    return tms.days() + 'd ' + tms.hours() + 'h ' + tms.minutes() + 'm ' + (tms.seconds() <= 9 ? '0' + tms.seconds() : tms.seconds()) + 's';
+  };
+  $('#countdown').html(countdownStr(tmp));
+  $.ajax({
+    url: 'time.php',
+    data: {
+      t: (new Date()).getTime()
+    },
+    success: function(resp) {
```
Comment (jszwedko) [inline js/waiting.js:13]: I think this should have an error handler as well though I don't have immediate ideas for good content to show the user.

File: time.php [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<?php
+  $data = [];
+  try {
+      $data = file_get_contents("https://digitalservices.sfgov.org/servertime.php");
+  } catch (Exception $ex) {
+      echo $ex;
```
Comment (jszwedko) [inline time.php:6]: I think it'd be better to log the exception here and either:

* Return structured JSON with the error message
* Return a `500`

**Review by jszwedko — COMMENTED** [review]
Overall 🎊 🎉  . Just some comments in-line.

File: js/waiting.js (hunk 5 above)
Comment (aekong) [reply js/waiting.js:18]: it's a nod to hui's style of coding back in the day.  i'll change it

File: css/waiting.css (hunk 1 above)
Comment (aekong) [reply css/waiting.css:1]: yeah, it's coming from designer.  i'll run it through a formatter

File: time.php (hunk 7 above)
Comment (aekong) [reply time.php:6]: yeah, good point.  did it in a hurry, forgot all about it

File: css/waiting.css (hunk 1 above)
Comment (aekong) [reply css/waiting.css:1]: fixed in commit: 4f8224a0174f523fece7fb8c9fad6bdc3d7fd920

File: index.php (hunk 2 above)
Comment (aekong) [reply index.php:9]: i better check that!

File: js/waiting.js (hunk 4 above)
Comment (aekong) [reply js/waiting.js:3]: addressed in commit: c1b63e5b775889446974e4509b87fa5a2dd2b7cb

File: js/waiting.js (hunk 5 above)
Comment (aekong) [reply js/waiting.js:18]: addressed in commit: c1b63e5b775889446974e4509b87fa5a2dd2b7cb

sorry @hshaosf

File: js/waiting.js (hunk 6 above)
Comment (aekong) [reply js/waiting.js:13]: addressed in commit: c1b63e5b775889446974e4509b87fa5a2dd2b7cb

sorry, lumped some things together in this commit

File: time.php (hunk 7 above)
Comment (aekong) [reply time.php:6]: addressed in commit: 21899593ee9b83eed4efe8465d58901db7031741

File: css/waiting.css (hunk 1 above)
Comment (jszwedko) [reply css/waiting.css:1]: 👍 it is beautiful

File: js/waiting.js (hunk 4 above)
Comment (jszwedko) [reply js/waiting.js:3]: 👍 much more readable, thanks!

File: time.php (hunk 7 above)
Comment (jszwedko) [reply time.php:6]: 👍

File: css/waiting.css (hunk 1 above)
Comment (hshaosf) [reply css/waiting.css:1]: I would prefer to avoid using external assets if possible, especially for something that requires fastest load time possible.

File: index.php (hunk 3 above)
Comment (aekong) [reply index.php:31]: addressed in commit: 81525e23e2377b35982ba0ceabfb4e50d10e1f82

File: css/waiting.css (hunk 1 above)
Comment (aekong) [reply css/waiting.css:1]: @hshaosf d1f86296c79dd3ad85a3876c8c43ed40af6efba3

File: index.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
+<?php
+header("Cache-Control: no-store, no-cache, must-revalidate");
```
Comment (jszwedko) [inline index.php:2]: Stumbled across this timely blog post https://www.fastly.com/blog/headers-we-dont-want

Apparently `Pragma` has no meaning in responses. According to the post `Cache-Control: no-cache, private` should be all you need to make sure the page isn't cached, but these extra headers likely shouldn't do harm.

File: index.php (hunk 2 above)
Comment (aekong) [reply index.php:9]: it matches _now_

**Comment by aekong** [comment]
fixed a bunch of things, pls close if they meet expectations 😃

**Comment by jszwedko** [comment]
Looks good to me! I'll close out, but please feel free to reopen if you have remaining comments.
